### ISSUE TITLE
`download-url` no longer working with public wikipedia URLs (asking for credentials)

### ISSUE THREAD

**Issue by mih** [datalad/datalad#5801]
```zsh
% datalad download-url https://upload.wikimedia.org/wikipedia/commons/a/a5/Flower_poster_2.jpg
[INFO   ] Downloading 'https://upload.wikimedia.org/wikipedia/commons/a/a5/Flower_poster_2.jpg' into '/tmp/some/' 
Authenticated access to https://upload.wikimedia.org/wikipedia/commons/a/a5/Flower_poster_2.jpg has failed.
Would you like to setup a new provider configuration to access url? (choices: [yes], no): n

ERROR: 
Interrupted by user while doing magic: 

# whereas
% wget https://upload.wikimedia.org/wikipedia/commons/a/a5/Flower_poster_2.jpg
--2021-07-20 13:47:28--  https://upload.wikimedia.org/wikipedia/commons/a/a5/Flower_poster_2.jpg
Resolving upload.wikimedia.org (upload.wikimedia.org)... 91.198.174.208, 2620:0:862:ed1a::2:b
Connecting to upload.wikimedia.org (upload.wikimedia.org)|91.198.174.208|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 4487679 (4.3M) [image/jpeg]
Saving to: ‘Flower_poster_2.jpg’

Flower_poster_2.jpg            100%[===================================================>]   4.28M  4.54MB/s    in 0.9s    

2021-07-20 13:47:29 (4.54 MB/s) - ‘Flower_poster_2.jpg’ saved [4487679/4487679]
```

This is happening in `master` and `maint`. Via @adswa

**Comment by adswa** [comment]
I don't understand the download-related code base at all, but when I am debugging this, I get into an FTP session. I also don't have any understanding of FTP sessions (haha), but would a HTTP request need an FTP session?

**Comment by adswa** [comment]
Mhhh, funnily enough, downgrading DataLad doesn't fix the issue. Could this be introduced by a different library? The last time I can confirm that the ``download-url`` command was working was when I last built the handbook. This is a while ago, in January, using datalad 0.14.0rc1.dev10. I have tried downgrading until datalad 0.13.5, but it keeps trying to ask for a provider configuration.

**Comment by adswa** [comment]
... or is it something about the URL? I tried with a random IKEA manual and it works...

```
datalad download-url -m "Really important manula" https://www.ikea.com/de/de/manuals/troxhult-wall-shelf-white__AA-2180177-2_pub.pdf
[INFO   ] Downloading 'https://www.ikea.com/de/de/manuals/troxhult-wall-shelf-white__AA-2180177-2_pub.pdf' into '/tmp/f/' 
download_url(ok): /tmp/f/troxhult-wall-shelf-white__AA-2180177-2_pub.pdf (file) 
add(ok): troxhult-wall-shelf-white__AA-2180177-2_pub.pdf (file)                 
save(ok): . (dataset)                                                           
action summary:                                                                 
  add (ok: 1)
  download_url (ok: 1)
  save (ok: 1)
```

**Comment by yarikoptic** [comment]
> , I get into an FTP session

hm... ftp support is done via requests_ftp which IIRC is a bit obscure . 

Re original issue - runing with `-l 1` we see

```
[DEBUG  ] http session: Creating brand new session 
[Level 5] Calling out into <bound method BaseDownloader._download of HTTPDownloader(authenticator=None, credential=None)> for https://upload.wikimedia.org/wikipedia/commons/a/a5/Flower_poster_2.jpg 
send: b'GET /wikipedia/commons/a/a5/Flower_poster_2.jpg HTTP/1.1\r\nHost: upload.wikimedia.org\r\nUser-Agent: python-requests/2.25.1\r\nAccept-Encoding: \r\nAccept: */*\r\nConnection: keep-alive\r\n\r\n'
reply: 'HTTP/1.1 403 Forbidden. Please comply with the User-Agent policy: https://meta.wikimedia.org/wiki/User-Agent_policy\r\n'
header: Date: Tue, 20 Jul 2021 12:46:31 GMT
header: Server: Varnish
header: X-Cache: cp1090 int
header: X-Cache-Status: int-front
header: Server-Timing: cache;desc="int-front", host;desc="cp1090"
header: Strict-Transport-Security: max-age=106384710; includeSubDomains; preload
header: Report-To: { "group": "wm_nel", "max_age": 86400, "endpoints": [{ "url": "https://intake-logging.wikimedia.org/v1/events?stream=w3c.reportingapi.network_error&schema_uri=/w3c/reportingapi/network_error/1.0.0" }] }
header: NEL: { "report_to": "wm_nel", "max_age": 86400, "failure_fraction": 0.05, "success_fraction": 0.0}
header: Permissions-Policy: interest-cohort=()
header: Set-Cookie: WMF-Last-Access=20-Jul-2021;Path=/;HttpOnly;secure;Expires=Sat, 21 Aug 2021 12:00:00 GMT
header: X-Client-IP: 76.24.253.1
header: Content-Type: text/html; charset=utf-8
header: Content-Length: 1904
header: Connection: keep-alive
[DEBUG  ] Authenticated access was denied:  
Authenticated access to https://upload.wikimedia.org/wikipedia/commons/a/a5/Flower_poster_2.jpg has failed.
Would you like to setup a new provider configuration to access url? (choices: [yes], no): 

```
so the

```HTTP/1.1 403 Forbidden. Please comply with the User-Agent policy: https://meta.wikimedia.org/wiki/User-Agent_policy\```

is the culprit -- we (or requests?) should look into providing a User-Agent

**Comment by adswa** [comment]
That clears it up a bit at least. But the policy is in place since 2010, and it fails only now? 

> hm... ftp support is done via requests_ftp which IIRC is a bit obscure .

Ah, I see, it monkeypatches the session...  I couldn't wrap my head around how in https://github.com/datalad/datalad/blob/88a05469b2698d3a93ff89aa3aa1e34e1871dd7b/datalad/downloaders/http.py#L528 ``requests.Session()`` leads me into  ``/home/adina/env/handbook2/lib/python3.9/site-packages/requests_ftp/ftp.py(18)``. But I don't get how this is triggered differently between the two urls I tried....

**Comment by yarikoptic** [comment]
FWIW, we do provide User-Agent string, example 
```
76.24.253.1 - - [20/Jul/2021:09:19:44 -0400] "GET /robots.txt HTTP/1.1" 200 1732 "-" "Wget/1.21"
76.24.253.1 - - [20/Jul/2021:09:19:53 -0400] "GET /robots.txt HTTP/1.1" 200 1732 "-" "python-requests/2.25.1"
```
not sure why wiki doesn't like this one...?

**Comment by christian-monch** [comment]
Because of the `-` in `python-requests` it seems. Look at this:

```
>>> import requests
>>> requests.get("https://upload.wikimedia.org/wikipedia/commons/a/a5/Flower_poster_2.jpg", headers={"User-Agent": "python-requests/2.23.0"})
<Response [403]>
>>> requests.get("https://upload.wikimedia.org/wikipedia/commons/a/a5/Flower_poster_2.jpg", headers={"User-Agent": "python/2.23.0"})
<Response [200]>
```

**Comment by yarikoptic** [comment]
I am getting PR ready for this one

**Comment by yarikoptic** [comment]
meanwhile I believe that wikipedia fixed that issue as well -- downloads without any tune ups, submitting a PR anyways.  Was looking into either we can channel also the informative part of the returned value, but it is not available as part of the exception we are receiving :-/

**Comment by adswa** [comment]
Closed by #5802.
